### PR TITLE
chore: update k8s to 1.16

### PR DESCRIPTION
**Pull request by jpfe-tid**
also:
- improve dockerfile
- update deps
  - go 1.13 -> 1.14
  - stretchr/testify 1.3.0 -> 1.6.1
  - go-yaml 2.2.8 -> 2.3.0
- switch to gomodules

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,17 @@
-FROM golang:1.13.5-alpine as build
+FROM golang:1.14.4-alpine3.12 as build
 
-RUN apk --no-cache add git curl
-RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+WORKDIR /root
+COPY . .
 
-RUN addgroup -g 10001 app && adduser -D -g '' -G app -s /bin/false -h /go/src/github.com/Telefonica/kube-graffiti -u 10001 app
```
Comment (jpfe-tid) [inline Dockerfile:6]: app user is not needed at build stage

File: Dockerfile [hunk 2]
```diff
@@ -1,22 +1,17 @@
-FROM golang:1.13.5-alpine as build
+FROM golang:1.14.4-alpine3.12 as build
 
-RUN apk --no-cache add git curl
```
Comment (jpfe-tid) [inline Dockerfile:3]: git and curl not needed anymore

File: Dockerfile [hunk 3]
```diff
@@ -1,22 +1,17 @@
...
-RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+WORKDIR /root
+COPY . .
 
-RUN addgroup -g 10001 app && adduser -D -g '' -G app -s /bin/false -h /go/src/github.com/Telefonica/kube-graffiti -u 10001 app
-WORKDIR /go/src/github.com/Telefonica/kube-graffiti
-
-COPY . /go/src/github.com/Telefonica/kube-graffiti
-RUN dep ensure -v
```
Comment (jpfe-tid) [inline Dockerfile:10]: switch to gomodules

dependencies are fetched automatically on any go command (run, test, build, etc...)

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	github.com/cameront/go-jsonpatch v0.0.0-20180223123257-a8710867776e
+	github.com/davecgh/go-spew v1.1.1
+	github.com/mitchellh/mapstructure v1.3.2
+	github.com/rs/zerolog v1.19.0
+	github.com/spf13/cobra v1.0.0
+	github.com/spf13/viper v1.7.0
+	github.com/stretchr/testify v1.6.1
+	github.com/xeipuuv/gojsonpointer v0.0.0-20190905194746-02993c407bfb // indirect
+	gopkg.in/yaml.v2 v2.3.0
+	k8s.io/api v0.16.11
```
Comment (jpfe-tid) [inline go.mod:15]: k8s.io uses 0.x versions to force disable semver upgrades

Comment (jpfe-tid) [reply go.mod:15]: e.g. this depencency corresponds to 1.16.11

**Comment by daguilarv** [comment]
Thanks for your great contribution!! We are currently using kube-graffiti in 1.16 k8s clusters. I guess when you made it compatible with 1.15 these changes are still valid. It would be great if you could write a mini how to about upgrading kube-graffiti to be compatible with higher kubernetes versions.

**Review by daguilarv — APPROVED** [review]
🎩

**Comment by jpfe-tid** [comment]
Thanks @daguilarv,

I sent this PR due to @palmerabollo request. He told me that kube-graffiti may not access apps/v1/deployments properly and it could be related to a version mismatch (1.15. vs 1.16).

Upgrading Kubernetes dependencies should be straightforward unless any function signature changes

E.g.:

```diff
- func foo(a string, b int) {}
+ func foo(x bool, a string, b int) {}
```

The dependencies to be update are:

- k8s.io/api
  - maps to https://github.com/kubernetes/api
- k8s.io/apimachinery
  - maps to https://github.com/kubernetes/apimachinery
- k8s.io/client-go
  - maps to https://github.com/kubernetes/client-go

Tags can be listed with `git ls-remote` or through GitHub Releases/Tags page.

E.g.:

```sh
$ git ls-remote https://github.com/kubernetes/api | grep tags | grep 'v0.16'
...
94c561cbdad18baa065c685c40361f5e2d691958	refs/tags/v0.16.11
b9332cd76d53ed3324232a1ea050323e7a9d1bb6	refs/tags/v0.16.11^{}
d8639d7bf6cacd44430cbabe42ed566dcb3a647c	refs/tags/v0.16.11-rc.0
0355febb3295aa8e6c0df8626adcb5289a4c22d7	refs/tags/v0.16.11-rc.0^{}
```

Get the desired tag and set it in `go.mod` file.

https://github.com/Telefonica/kube-graffiti/blob/hore/update-k8s-1-16/go.mod#L15-L17

**Comment by palmerabollo** [comment]
Thanks @jpfe-tid.
Regarding Diego's comment, I think this component should have a different release for each k8s version (e.g. the approach followed by cluster-autoscaler).

**Comment by jpfe-tid** [comment]
@palmerabollo I agree. Original repo seems to be abandoned. If this PR is merged, I'll tag `1.16.0` and write in the release notes that this release uses kubernetes 1.16 dependencies.

**Comment by jpfe-tid** [comment]
PD: additional repo cleanup is needed

**Review by palmerabollo — APPROVED** [review]

**Comment by jpfe-tid** [comment]
https://github.com/Telefonica/kube-graffiti/releases/tag/1.16.0

https://hub.docker.com/layers/telefonica/kube-graffiti/1.16.0/images/sha256-243cdd8dd15708478f6cbbc2ea43c982f46ec67e48fdaecb6dd3d3ae6056b1f5?context=explore

**Pull request merged**